src/main/java/net/jonathangiles/tools/teenyhttpd/model/TypedResponse.java

```diff
@@ -0,0 +1,38 @@
+package net.jonathangiles.tools.teenyhttpd.model;
+
+import net.jonathangiles.tools.teenyhttpd.implementation.TypedResponseImpl;
+
+public interface TypedResponse<T> extends Response {
+
+    T getValue();
+
+    TypedResponse<T> header(String key, String value);
```

Comment:
Do we need this? There is already setHeader on Response

Reply:
Same as I thought, but `java.net.http.HttpRequest` also have `header` & `setHeader` methods, That's why I thought there was no issue, I can remove `header` since is pure sugar syntax


Comparative

```java
return ok(new Pet(name, age, type))
                .setHeader(new Header("Authorization", auth))
```

```java
return ok(new Pet(name, age, type))
                .header("Authorization", auth);
```

Or I can rename `header` method to `setHeader`

```java
return ok(new Pet(name, age, type))
                .setHeader(new Header("Authorization", auth))
```

```java
return ok(new Pet(name, age, type))
                .setHeader("Authorization", auth);
```

Reply:
Let's assume you rename header to setHeader. There is a question over what the semantics of it mean - because headers can have multiple values, does calling this add the value if the key already exists, or does it replace all existing values? Presumably the user expectation is to add.... But then how do they replace? We just need to be consistent here, but again I'm happy to have syntactic sugar for common scenarios

Reply:
I renamed `header` to `setHeader` in the last commit, (I just read JLBP-1 at your blog ) the next commit I'll add `setHeader(String..)`

```java
setHeader(String key, String ... values)
```

This method will also replace all the existing values as the existing one

Reply:
We can also consider addHeader API for the add use cases